scripts/test

```diff
@@ -0,0 +1,75 @@
+#!/bin/sh
+#
+# Run all tests.
+#
+
+set -uC
+cd "$(dirname "$0")"
+
+### Setup Vim and other dependencies.
+#####################################
+out=$(./install-vim "${1:-}" 2>&1)
+if [ $? -ne 0 ]; then
+  echo "$out"
+  exit 4
+fi
+
+vim=$1
+vimdir=$(echo "$out" | tail -n1)
+vimgo="$(dirname "$(readlink -f .)")" # TODO: doesn't work on OSX I think.
+
+# Make sure all Go tools and other dependencies are installed.
+./run-vim $vim +':silent :GoInstallBinaries' +':qa'
+[ -d "$vimdir/share/vim/vimgo/pack/vim-go/start/vim-vimhelplint" ] || \
+  git clone --quiet https://github.com/machakann/vim-vimhelplint \
+    "$vimdir/share/vim/vimgo/pack/vim-go/start/vim-vimhelplint"
```

Comment:
Don't think these should be part of the test. Without this I should still run the tests on macOS.

Reply:
I'm afraid that I don't quite follow your comment. What do you mean with "this"? Cloning vimhelplint or the `:GoInstallBinaries` (or both?)

Reply:
This is the whole section above that installs Vim on the machine. Test should just run the tests, it shouldn't try to install anything every time I execute it.

Reply:
Okay; how would you do it? Have the script fail if the vim in `/tmp/vim-go-test/...` doesn't exist?

Reply:
One of the reasons I like this approach is that it's very easy to run tests: just run `./bin/test vim-7.4` and presto! This makes it easy for people to actually start *writing* some tests.

It will cache the results so only the first time takes a few minutes, and outside of the `:GoInstallBinaries` it shouldn't modify anything outside of the `/tmp/vim-go/test/` directory.

Maybe we could split it up in to `./bin/setup` and `./bin/run-test` or some such? Personally I don't really care as such, as long as running tests remains easy and "fool-proof" with just a few commands.

Reply:
I'm in favor of splitting it. Maybe `./bin/install` is a better name. We have makefile where we can combine both.

Reply:
Alright, I'll work on that later :+1:

Reply:
Perhaps we should run the tests in a Docker container so that the user's machine/environment is never polluted with what tests do?

Reply:
> Perhaps we should run the tests in a Docker container so that the user's machine/environment is never polluted with what tests do?

Yeah, could add that. It would largely be the same scripts, but with a Docker wrapper.
Personally I've had a lot of problems running Docker on my desktop machine and would prefer to avoid it.